golem-worker-executor/src/durable_host/io/poll.rs

```diff
@@ -56,6 +57,29 @@ impl<Ctx: WorkerCtx> HostPollable for DurableWorkerCtx<Ctx> {
 
 impl<Ctx: WorkerCtx> Host for DurableWorkerCtx<Ctx> {
     async fn poll(&mut self, in_: Vec<Resource<Pollable>>) -> anyhow::Result<Vec<u32>> {
+        // check if all pollables are promise backed. In this case we can suspend immediately
```

Comment:
Interesting. I expected this will have to be somehow supported in the underlying wasmtime-wasi poll in our fork. Maybe the "suspend for sleep" thing can also be implemented in a way that we can revert the fork to the its original state - but let's not deal with that now